src/sagemaker/workflow/airflow.py

```diff
@@ -29,7 +29,7 @@ def prepare_framework(estimator, s3_operations):
     """
     if estimator.code_location is not None:
         bucket, key = fw_utils.parse_s3_url(estimator.code_location)
-        key = os.path.join(key, 'source', 'sourcedir.tar.gz')
+        key = os.path.join(key, estimator._current_job_name, 'source', 'sourcedir.tar.gz')
```

Comment:
iirc we have a potential bug here where `_current_job_name` isn't set if an estimator is created using `attach`

Reply:
After more thoughts, I think it's more than this PR. We can have a new design and PR for attach in this case.